spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkSQLProperties.java

```diff
@@ -42,4 +42,9 @@ private SparkSQLProperties() {}
   // Controls whether to check the order of fields during writes
   public static final String CHECK_ORDERING = "spark.sql.iceberg.check-ordering";
   public static final boolean CHECK_ORDERING_DEFAULT = true;
+
+  // Controls whether to preserve the existing grouping of data while planning splits
+  public static final String PRESERVE_DATA_GROUPING =
```

Comment:
There were multiple names I considered and `spark.sql.iceberg.split.preserve-partition-boundaries` was one of them. I discarded it because we are not really preserving partition boundaries if there are multiple specs. Also, Spark will push down join keys in the future so we won't really respect partition boundaries as such.

That being said, I'll think more tomorrow.

Reply:
Name is fine to me, Ideally this isn't something that get's actually configured by the end user

Reply:
I am afraid this one will be pretty public. Users will have to explicitly enable this as we don't know if Spark can benefit from the reported distribution and skip shuffles.

Reply:
I think this name makes sense.